docs/_posts/2019-08-06-your-regex-here.md

```diff
@@ -0,0 +1,614 @@
+* `Dash_Punctuation` or `Pd`
+* `Open_Punctuation` or `Ps`
+* `Close_Punctuation` or `Pe`
+* `Initial_Punctuation` or `Pi`
+* `Final_Punctuation` or `Pf`
+* `Other_Punctuation` or `Po`
+* `Symbol` or `S`
+* `Math_Symbol` or `Sm`
+* `Currency_Symbol` or `Sc`
+* `Modifier_Symbol` or `Sk`
+* `Other_Symbol` or `So`
+* `Separator` or `Z`
+* `Space_Separator` or `Zs`
+* `Line_Separator` or `Zl`
+* `Paragraph_Separator` or `Zp`
+* `Other` or `C`
+* `Control or cntrl` or `Cc`
+* `Format` or `Cf`
+* `Surrogate` or `Cs`
+* `Private_Use` or `Co`
+* `Unassigned` or `Cn`
+
+Again, these are directly from the Unicode specification.
+
+So, you might match a title-case string as such:
+
+    token titlestring { <:Titlecase_Letter> <:Lowercase_Letter>* }
+
+To negate a property, simply follow the colon with a `!`.
+
+## Character classes
+
+So far, these have all been expressed as "sub-rules". But you can
+also use them in "character class" constructions that can
+only ever match a single character and can be built up in pieces.
+
+Character classes can be explicit lists of characters:
+
+    <[a..z]> # lower ascii alpha
+
+Or they can be constructed from other classes:
+
+    <+<alpha>-<:Uppercase_Letter>>
+
+Which is to say, "the alpha class, but excluding the upper case
+letters".
+
+Unicode codepoints can also be referenced by hexadecimal number:
+
+    \x1234 # always four digits
+
+or:
+
+    \x[12345]
+
+Or by name:
+
+    \c[GREEK SMALL LETTER ALPHA]
+
+## Quantifiers
+
+The typical regular expression quantifiers are supported:
+
+* `+` - Match the preceding "atom" (single matching literal, group, sub-rule
+  or class) one or more times.
+* `*` - Match the preceding atom zero or more times.
+* `** min[..max]` - Match the preceding atom min to max (\* means forever)
+  times.
+
+These constructs modify preceding quantifiers:
+
+* `?` - Match non-greedily
+* `!` - Match greedily (the default)
+* `:` - Do not perform backtracking for the previous quantifier
+* `%` - Match the preceding atom _with quantifier_ separated by the following
+  atom (e.g. `\w+ % ','` which matches comma-separated words)
+* `%%` - Same as `%`, but allows a trailing separator.
+
+And finally, there is a special syntax for matching balanced expressions:
+
+    '(' ~ ')' <expression>
+
+Which matches the sub-rule `<expression>` enclosed in parentheses.
+
+This is not substantially different from:
+
+    '(' <expression> ')'
+
+But it does allow the system to provide more useful diagnostics
+because it knows that the intent of the user was to match a
+balanced beginning and ending token.
+
+## Conjunctions
+
+The `|` of most regular expression syntaxes is now `||`. This will match
+on the first matching alternative between the left and right hand sides.
+
+    'a' || 'aa' # will never match the right
+
+While the `|` now matches the longest of the two, as most BNF-like
+parser specifications expect, so:
+
+    'a' | 'aa' # will only ever match the right
```

Comment:
If the string is `a` or `ab`, the left also matches.

Better wording would be "matches the right if the input string is `aa`".

Reply:
My head cold got to me. Good call. I'll fix.